sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/package.scala

```diff
@@ -204,6 +204,27 @@ package object util extends Logging {
     truncatedString(seq, "", sep, "", maxFields)
   }
 
+  /** Whether we have warned about plan string truncation yet. */
+  private val planSizeWarningPrinted = new AtomicBoolean(false)
+
+  def withSizeLimitedWriter[T](writer: Writer)(f: (Writer) => T): Option[T] = {
+    try {
+      // Subtract 3 from the string length to leave room for the "..."
+      val limited = new SizeLimitedWriter(writer, SQLConf.get.maxPlanStringLength - 3)
+      Some(f(limited))
+    }
+    catch {
+      case e: WriterSizeException =>
+        writer.write("...")
+        if (planSizeWarningPrinted.compareAndSet(false, true)) {
+          logWarning(
+            "Truncated the string representation of a plan since it was too long. This " +
```

Comment:
If it is possible I would print out the size of the plan. So, an user could know how much he/she should bump the SQL config.

Reply:
That should be possible.